### PR TITLE
Document `foreignKey` option on belongsToMany & `where` for include

### PR DESCRIPTION
**Pull request by bulkan**

### CODE REVIEW

**Comment by janmeier** [comment]
Sweet :)

Could you mention include.required as well? By default when where is set the query will only return instances of the parent model which has a matching included model - setting required to false will return all parent models, but only those children matching the where

**Comment by bulkan** [comment]
@janmeier If I understand the [API docs](http://sequelize.readthedocs.org/en/latest/api/model/#findalloptions-promisearrayinstance) when `include.where` is set then `include.required` is implicitly set to `true` ?

**Comment by janmeier** [comment]
Yep

**Comment by bulkan** [comment]
@janmeier updated branch

File: docs/docs/models-definition.md [hunk 1]
```diff
@@ -1075,6 +1075,51 @@ User.findAll({ include: [{ model: Tool, as: 'Instruments' }] }).then(function(us
 })
 ```
 
+When eager loading we can also filter the associated model using `where`. This will return all
```
Comment (janmeier) [inline docs/docs/models-definition.md]: Hmm, it appears theres some duplication in models usage and models definition - eager loading should be n usage

**Comment by janmeier** [comment]
thanks @bulkan I'll reorganize it a bit later :)

**Pull request merged**